sanho/spring/src/main/java/umc/spring/service/store/StoreQueryServiceImpl.java

```diff
@@ -0,0 +1,61 @@
+package umc.spring.service.store;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.data.domain.Page;
+import umc.spring.domain.Mission;
+import umc.spring.domain.Review;
+import umc.spring.domain.Store;
+import umc.spring.repository.mission.MissionRepository;
+import umc.spring.repository.review.ReviewRepository;
+import umc.spring.repository.store.StoreRepository;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+@Transactional(readOnly = true) // 조회에만
+public class StoreQueryServiceImpl implements StoreQueryService{
+
+    private final StoreRepository storeRepository;
+    private final ReviewRepository reviewRepository;
+    private final MissionRepository missionRepository;
+
+    @Override
+    public Optional<Store> findStore(Long id) {
+        return storeRepository.findById(id);
+    }
+
+    @Override
+    public List<Store> findStoresByNameAndScore(String name, Float score) {
+        List<Store> filteredStores = storeRepository.dynamicQueryWithBooleanBuilder(name, score);
+
+        filteredStores.forEach(store -> System.out.println("Store: " + store));
+
+        return filteredStores;
+    }
+
+    // week9 실습 코드
+    @Override
+    public Page<Review> getReviewList(Long storeId, Integer page) {
+        Store store = storeRepository.findById(storeId).get(); // 먼저 storeId로 가게를 찾고
+
+//        Store store = storeRepository.findById(storeId)
+//                .orElseThrow(() -> new StoreHandler(ErrorStatus.STORE_NOT_FOUND));
+
+        Page<Review> storePage = reviewRepository.findAllByStore(store, PageRequest.of(page, 10)); // 해당 가게의 리뷰 10개 단위로 가져오기
+        // Page<Review> : 데이터, 페이지 번호, 페이지 개수, 전체 데이터 개수, 페이지 크기, 페이지 여부, 정렬 정보 등
+
+        return storePage;
+    }
+
+    // week9 미션 - 특정 가게 미션 목록 조회
+
+    @Override
+    public Page<Mission> getMissionList(Long storeId, Integer page) {
+        Store store = storeRepository.findById(storeId).get();

```

Comment:
현재 코드는 `Optional`의 `.get()` 메서드를 사용하여 `storeId`로 가게를 조회하고 있습니다:
이 방식은 가게가 존재하지 않을 경우 NoSuchElementException을 발생시킬 수 있으므로 적절한 예외 처리가 필요합니다.
orElseThrow() 메서드를 사용해 명시적인 예외를 던지거나, Optional이 비어 있는 경우에 대한 처리를 추가하면 안전성을 높일 수 있습니다:

```suggestion
        Store store = storeRepository.findById(storeId)
            .orElseThrow(() -> new StoreHandler(ErrorStatus.STORE_NOT_FOUND));
```